nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,826 @@
+  & \rm{agg}(1,4), \rm{agg}(-2,-6), \rm{agg}(3,1) &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & \rm{agg}(4,1), \rm{agg}(-3,2), \rm{agg}(2,3) & \\
+  & \rm{agg}(-4,1), \rm{agg}(3,-2), \rm{agg}(2,3) &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & \rm{agg}(5,4), \rm{agg}(-1,-3), \rm{agg}(3,2) &\\
+  & \rm{agg}(-5,-4), \rm{agg}(1,3), \rm{agg}(3,2) &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+
+\xrightarrow[]{\rm{argmin}}
+
+\begin{bmatrix}
+
+```
+\begin{bmatrix}
+  & 0,0, 1 & \\
+  & 0, 1,1 &
+\end{bmatrix}
+\\ 
+\begin{bmatrix} 
+  & 1, 0,0 & \\
+  & 0,1,0 &
+\end{bmatrix}
+\\ 
+\begin{bmatrix}
+  & 1, 1,1 &\\
+  & 0, 0,1 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+$$
+
+```elixir
+Nx.argmin(t, axis: :x)
+```
+
+###### Aggregate along the second axis `axis: :y` (equivalently `axis: 1`) .
+
+The axis `y` will collapse and the shape of the resultant tensor `r` is `[x: 2, z: 2, t: 3]`.
+
+How? The axis $x$ is before the selected one $y$ so it will remain. We therefor apply the procedure as above for _each_ sub $x$-slice.
+
+More precisely, fix the $x$-slice,  and let's consider the $x=1$ one. You collect in each sub `y`-slice the elements with the same remaining indexes "on the right", $z,t$ (and $x$ of course). You apply the aggregation and the result will be the value at location $r[1][z][t]$ of the resultant tensor $r$. So the elements of the slice $x=1$ of the resultant tensor $r$ will be:
+
+$$
+r[x_{=1}][z][t] = \rm{agg}\big([t[x_{=1}][0][z][t], t[x_{=1}][1][z][t], t[x_{=1}][2][z][t]\big)
+$$
+
+You repeat this operation for each $x$-slice.
+
+In the example below, we used again the `argmin` function.
+
+$$
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & \boxed{1}, 2, 3, & \\
+  & 1, -2, 3, &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & \boxed{4}, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & \boxed{5}, -1, 3 &\\
+  & -5, 1, 3 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+  \\
+  x=1
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 4, 6, 1 & \\
+  & 4, \boxed{-6}, 1 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & 1, \boxed{-2}, 3 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, \boxed{3}, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+
+\to
+  \begin{bmatrix} x=0
+
+```
+\begin{bmatrix}
+  &\boxed{\rm{agg}(1,4,5)}, \rm{agg}(2,-3,-1), \rm{agg}(3, 2,3)& \\
+  & \rm{agg}(1,-4,-5), \rm{agg}(-2,3,1), \rm{agg}(3,2,3) &
+\end{bmatrix}
+\\ 
+x=1
+\begin{bmatrix} 
+  & \rm{agg}(4,1,4), \rm{agg}(6,2,-3), \rm{agg}(1,3,2) & \\
+  & \rm{agg}(4,1,-4), \boxed{\rm{agg}(-6,-2,3)}, \rm{agg}(1,3,2) &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+
+\xrightarrow[]{\rm{argmin}}
+
+\begin{bmatrix}
+
+```
+\begin{bmatrix}
+  & 0, 2, 1 & \\
+  & 2, 0,1 &
+\end{bmatrix}
+\\ 
+\begin{bmatrix} 
+  & 1, 2,0 & \\
+  & 2,0,0 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+$$
+
+```elixir
+Nx.argmin(t, axis: 1)
+```
+
+###### Aggregate along the `axis: :z` axis (equivalently `axis: 2`).
```

Comment:
I don't think showing the aggregation over the next 2 axes brings anything significant to the table.
If you want to show like axes 0 and 2, I'm ok with it, but showing all 4 of them doesn't seem to add anything after the second example.